src/components/DynamicFormContainer/DynamicFormLayout.tsx

```diff
@@ -1,53 +1,131 @@
+import Ajv from "ajv";
 import React, { FunctionComponent, useState } from "react";
 import { Redirect } from "react-router";
-import { useActiveFormContext } from "../../common/context/activeForm";
 import { useConfigContext } from "../../common/context/config";
-import { SvgIcon, SvgIconArrowLeft } from "../../UI/SvgIcon";
-import { Title } from "../../UI/Title";
+import { useFormsContext } from "../../common/context/forms";
+import { Button } from "../../UI/Button";
+import { SvgIcon, SvgIconTrash, SvgIconXCircle } from "../../UI/SvgIcon";
 import { ToggleSwitch } from "../../UI/ToggleSwitch";
 import { Container } from "../Container";
-import { ProgressBar } from "../ProgressBar";
+import { DeleteModal } from "./DeleteModal";
 import { DynamicForm } from "./DynamicForm";
+import { DynamicFormHeader } from "./DynamicFormHeader";
 
 export const DynamicFormLayout: FunctionComponent = () => {
   const { config } = useConfigContext();
+  const [showDeleteModal, setDeleteModal] = useState(false);
   const [isPreviewMode, setIsPreviewMode] = useState(false);
-  const { activeFormIndex, setActiveFormIndex } = useActiveFormContext();
-  if (activeFormIndex === undefined) return <Redirect to="/forms-selection" />;
-  const activeForm = config?.forms[activeFormIndex];
-  if (!activeForm) return <Redirect to="/forms-selection" />;
+  const {
+    forms,
+    setForms,
+    setActiveFormIndex,
+    currentForm,
+    setCurrentFormData,
+  } = useFormsContext();
+  const [isSubmitted, setIsSubmitted] = useState(false);
+  const [formError, setFormError] = useState<Ajv.ErrorObject[] | null | undefined>();
```

Comment:
```suggestion
  const [formError, setFormError] = useState<Ajv.ErrorObject[] | null>(null);
```

I'm assuming that the state will be null when there is no error? If that's the case can initialise to null instead of letting it default to undef, then you will have 2 types of states, instead of 3.

Reply:
ok

Reply:
i tried, the ajv.errors keeps throwing the error, it needs the undefined.